spec/factories/chargeback_rate.rb

```diff
@@ -2,13 +2,54 @@
   factory :chargeback_rate do
     guid                   { MiqUUID.new_guid }
     sequence(:description) { |n| "Chargeback Rate ##{n}" }
-    rate_type              'Compute'
-  end
+    rate_type 'Compute'
+
+    trait :with_details do
+      chargeback_rate_details do
+        [FactoryGirl.create(:chargeback_rate_detail_memory_allocated, :tiers_with_three_intervals),
+         FactoryGirl.create(:chargeback_rate_detail_memory_used, :tiers)]
+      end
+    end
+
+    transient do
+      per_time 'hourly'
+    end
```

Comment:
Move this above any `trait`s so it's easier to find and doesn't get lost in the body of this file.